src/main/puzzle/combo-tracker.gd

```diff
@@ -0,0 +1,98 @@
+class_name ComboTracker
+extends Node
+"""
+Keeps track of the player's combo.
+
+Increments the combo when the player does well, breaks the combo when the player messes up.
+"""
+
+signal combo_break_changed(value)
+
+# bonus points which are awarded as the player continues a combo
+const COMBO_SCORE_ARR = [0, 0, 5, 5, 10, 10, 15, 15, 20]
+
+# number of lines the player has cleared without dropping their combo
+var combo := 0
+
+# The number of pieces the player has dropped without clearing a line or making a box.
+var combo_break := 0
+
+# 'true' if the player drops a piece which continues the combo (typically, making a box or clearing a line)
+var piece_continued_combo := false
+
+# 'true' if the player drops a piece which discontinues the combo.
+# Some scenarios might break your combo if you clear a vegetable line, for example.
+var piece_broke_combo := false
+
+func _ready() -> void:
+	PuzzleScore.connect("game_prepared", self, "_on_PuzzleScore_game_prepared")
+	PuzzleScore.connect("game_ended", self, "_on_PuzzleScore_game_ended")
+
+
+func add_combo(y: int, total_lines: int, remaining_lines: int, box_ints: Array) -> void:
```

Comment:
Add method comment

Reply:
Added comment and renamed method to `add_combo_and_score` to be more self-documenting.